tests/unit/lms/services/oauth_http_test.py

```diff
@@ -71,3 +152,13 @@ def test_it(
     @pytest.fixture
     def OAuthHTTPService(self, patch):
         return patch("lms.services.oauth_http.OAuthHTTPService")
+
+
+@pytest.fixture(autouse=True)
+def OAuthTokenResponseSchema(patch):
+    return patch("lms.services.oauth_http.OAuthTokenResponseSchema")
+
+
+@pytest.fixture
+def oauth_token_response_schema(OAuthTokenResponseSchema):
+    return OAuthTokenResponseSchema.return_value
```

Comment:
I think both of these fixtures are local to `TestOAuthHTTPService`. Also I believe we normally put our `autouse` patches after other fixtures.

Reply:
Our usual style is to make patch fixtures be top-level functions. So for example if the module under test contains:

```python
from lms.validation.authentication import OAuthTokenResponseSchema
```

then the corresponding test module will contain a module-level function (not a class methods) like this:

```python
@pytest.fixture(autouse=True)
def OAuthTokenResponseSchema(patch):
    return patch("lms.services.oauth_http.OAuthTokenResponseSchema")
```

The reason is that `OAuthTokenResponseSchema` is being imported into the `oauth_http.py` _module_'s namespace. All code in `oauth_http.py` has access to the name `OAuthTokenResponseSchema`. So we want to make sure that all `oauth_http_test.py` tests are patching `OAuthTokenResponseSchema`.

When we put a convenience fixture like `oauth_token_response_schema()` that just returns `OAuthTokenResponseSchema.return_value` we usually put the convenience fixture right after the patch fixture that it depends on.

I didn't know we had a style of putting all of our `autouse` `patch` fixtures after other fixtures. I can see that being useful to group the `patch` fixtures together. But I wonder if convenience fixtures like `oauth_token_response_schema()` should be allowed to go in-between the patch fixtures, since it makes sense for them to be attached to the fixture they depend on.

Reply:
I'm pretty sure you've asked me to move stuff like this after all other fixtures. If this is not a thing anymore that would be good as it's one less thing to worry about. I suppose perhaps there's an exception for this particular pattern of adding another fixture to proxy `return_value`?

I'm pretty 50/50 on the utility of things like this where we can use `OAuthTokenResponseSchema.return_value` directly in tests instead.